src/main/scala/com/github/codelionx/distod/actors/worker/CandidateValidation.scala

```diff
@@ -0,0 +1,122 @@
+package com.github.codelionx.distod.actors.worker
+
+import com.github.codelionx.distod.actors.worker.CandidateValidation.{SplitCandidateValidationResult, SwapCandidateValidationResult}
+import com.github.codelionx.distod.partitions.{FullPartition, StrippedPartition}
+import com.github.codelionx.distod.types.{CandidateSet, OrderDependency}
+import com.github.codelionx.distod.types.OrderDependency.{ConstantOrderDependency, EquivalencyOrderDependency}
+
+import scala.collection.mutable
+
+
+object CandidateValidation {
+
+  case class SplitCandidateValidationResult(
+      validOds: Seq[OrderDependency],
+      removedCandidates: CandidateSet
+  )
+
+  case class SwapCandidateValidationResult(
+      validOds: Seq[OrderDependency],
+      removedCandidates: Seq[(Int, Int)]
+  )
+}
+
+
+trait CandidateValidation {
+
+  def checkSplitCandidates(
+      candidateId: CandidateSet,
+      splitCandidates: CandidateSet,
+      allAttributes: Seq[Int],
+      errors: Map[CandidateSet, Double]
+  ): SplitCandidateValidationResult = {
+    val errorCompare = errors(candidateId)
+    val validConstantODs = for {
+      a <- splitCandidates.unsorted
+      fdContext = candidateId - a
+      if errors(fdContext) == errorCompare // candidate fdContext: [] -> a holds
+    } yield fdContext -> a
+
+    val constantOds = validConstantODs.map {
+      case (context, attribute) => ConstantOrderDependency(context, attribute)
+    }.toSeq
+
+    val removedCandidates = {
+      val validCandidateSet = CandidateSet.fromSpecific(constantOds.map(_.constantAttribute))
+      if (validCandidateSet.nonEmpty)
+        validCandidateSet union (CandidateSet.fromSpecific(allAttributes) diff candidateId)
+      else
+        CandidateSet.empty
+    }
+    SplitCandidateValidationResult(
+      validOds = constantOds,
+      removedCandidates = removedCandidates
+    )
+  }
+
+  def checkSwapCandidates(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4581953)